src/librustc/ty/layout.rs

```diff
@@ -1513,6 +1531,59 @@ impl<'a, 'gcx, 'tcx> Layout {
             }
         }
     }
+
+    pub fn field_offset(&self,
```

Comment:
Seems like some comments, or better choice of names, would be good here. e.g. what is `i` -- the index of the field, I guess? `variant_index` is for structs/enums? (I'd also find the opposite order of those parameters more intuitive, but I can't say why -- I guess because the `i` is defined relative to `variant_index`)